Utilities/DataFlow/src/SubframeBuilderDevice.cxx

```diff
@@ -18,63 +18,88 @@
-// is still permitted, but deprecated. Some compilers do not implement this standard
-// correctly. It also has to be noticed that this error does not occur for all the
-// other public constexpr members
-constexpr uint32_t o2::DataFlow::SubframeBuilderDevice::mOrbitsPerTimeframe;
-constexpr uint32_t o2::DataFlow::SubframeBuilderDevice::mOrbitDuration;
-constexpr uint32_t o2::DataFlow::SubframeBuilderDevice::mDuration;
-
 using HeartbeatHeader = o2::Header::HeartbeatHeader;
 using HeartbeatTrailer = o2::Header::HeartbeatTrailer;
 using DataHeader = o2::Header::DataHeader;
+using SubframeId = o2::dataflow::SubframeId;
 
 o2::DataFlow::SubframeBuilderDevice::SubframeBuilderDevice()
   : O2Device()
 {
-  LOG(INFO) << "o2::DataFlow::SubframeBuilderDevice::SubframeBuilderDevice " << mDuration << "\n";
 }
 
 o2::DataFlow::SubframeBuilderDevice::~SubframeBuilderDevice()
 = default;
 
 void o2::DataFlow::SubframeBuilderDevice::InitTask()
 {
-//  mDuration = GetConfig()->GetValue<uint32_t>(OptionKeyDuration);
+  mOrbitDuration = GetConfig()->GetValue<uint32_t>(OptionKeyOrbitDuration);
+  mOrbitsPerTimeframe = GetConfig()->GetValue<uint32_t>(OptionKeyOrbitsPerTimeframe);
   mIsSelfTriggered = GetConfig()->GetValue<bool>(OptionKeySelfTriggered);
   mInputChannelName = GetConfig()->GetValue<std::string>(OptionKeyInputChannelName);
   mOutputChannelName = GetConfig()->GetValue<std::string>(OptionKeyOutputChannelName);
+  mFLPId= GetConfig()->GetValue<size_t>(OptionKeyFLPId);
+  mStripHBF= GetConfig()->GetValue<bool>(OptionKeyStripHBF);
 
+  // depending on whether the device is self-triggered or expects input,
+  // the handler function needs to be registered or not.
+  // ConditionalRun is not called anymore from the base class if the
+  // callback is registered
   if (!mIsSelfTriggered) {
-    // depending on whether the device is self-triggered or expects input,
-    // the handler function needs to be registered or not.
-    // ConditionalRun is not called anymore from the base class if the
-    // callback is registered
     LOG(INFO) << "Obtaining data from DataPublisher\n";
+    // Now that we have all the information lets create the policies to do the 
+    // payload extraction and merging and create the actual PayloadMerger.
+
+    // We extract the timeframeId from the number of orbits.
+    // FIXME: handle multiple socket ids
+    Merger::IdExtractor makeId = [this](std::unique_ptr<FairMQMessage> &msg) -> SubframeId {
+      HeartbeatHeader *hbh = reinterpret_cast<HeartbeatHeader*>(msg->GetData());
+      SubframeId id = {.timeframeId = hbh->orbit / this->mOrbitsPerTimeframe,
+                       .socketId = 0 };
+      return id;
+    };
+
+    // We extract the payload differently depending on wether we want to strip
+    // the header or not.
+    Merger::PayloadExtractor payloadExtractor = [this](char **out, char *in, size_t inSize) -> size_t {
+      if (!this->mStripHBF) {
+        *out = in;
+        return inSize;
```

Comment:
how about `return Merger::identity(...)` or the respective function name if it is renamed